src/fetchtastic/download/base.py

```
@@ -117,6 +127,223 @@ def download(self, url: str, target_path: Pathish) -> bool:
+                        ):
+                            await f.write(chunk)
+                            downloaded += len(chunk)
+
+                            if progress_callback:
+                                try:
+                                    result = progress_callback(
+                                        downloaded, total_size or None, target.name
+                                    )
+                                    if asyncio.iscoroutine(result):
+                                        await result
+                                except Exception as cb_err:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)